packages/platforms/celestica/x86-64/x86-64-cel-redstone-xp/onlp/builds/src/module/src/sys_eeprom.c

```diff
@@ -0,0 +1,66 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <unistd.h>
+#include <fcntl.h>
+
+#include "sys_eeprom.h"
+#include "cpld.h"
+#include "i2c_chips.h"
+#include "i2c_dev.h"
+
+static inline u_int8_t is_valid_tlvinfo_header(tlvinfo_header_t *hdr)
+{
+    int max_size = TLV_TOTAL_LEN_MAX;
+
+    return((strcmp(hdr->signature, TLV_INFO_ID_STRING) == 0) &&
+                            (hdr->version == TLV_INFO_VERSION) &&
+                            (be16_to_cpu(hdr->totallen) <= max_size));
+}
+
+int
+read_sys_eeprom(void *eeprom_data, int offset, int len)
+{
+  int ret = 0;
+  int i = 0;
+  u_int8_t *c;
+
+  int addr = SYS_EEPROM_OFFSET + offset;
+  unsigned char dev_id = 5;
+
+  c = eeprom_data;
+  if (eeprom_enable(dev_id) < 0) {
+    printf("ERROR: Cannot open I2C device\n");
+    return -1;
+  }
+
+  for (i = 0; i < len; i++) {
+    unsigned char buf;
+    ret = eeprom_read_byte(CONFIG_SYS_I2C_EEPROM_ADDR, addr, &buf);
+    *c = buf;
+    c++;
+    addr++;
+  }
+
+  eeprom_disable(dev_id);
+  return ret;
+}
+
+int
+read_eeprom(u_int8_t *eeprom, int *size)
+{
+    int ret;
+    tlvinfo_header_t *eeprom_hdr = (tlvinfo_header_t *) eeprom;
```

Comment:
Are these the ONIE TLVs? ONIE TVL buffer decoding is already builtin to the ONLPLIB module:
https://github.com/opencomputeproject/OpenNetworkLinux/blob/master/packages/base/any/onlp/src/onlplib/module/inc/onlplib/onie.h